src/diagrams/common/common.ts

```diff
@@ -0,0 +1,147 @@
+import DOMPurify from 'dompurify';
+import { MermaidConfig } from 'types/config';
+
+/**
+ * Gets the rows of lines in a string
+ *
+ * @param {string | undefined} s The string to check the lines for
+ * @returns {string[]} The rows in that string
+ */
+export const getRows = (s?: string): string[] => {
+  if (!s) return [''];
+  const str = breakToPlaceholder(s).replace(/\\n/g, '#br#');
+  return str.split('#br#');
+};
+
+/**
+ * Removes script tags from a text
+ *
+ * @param {string} txt The text to sanitize
+ * @returns {string} The safer text
+ */
+export const removeScript = (txt: string): string => {
+  return DOMPurify.sanitize(txt);
+};
+
+const sanitizeMore = (text: string, config: MermaidConfig) => {
+  // TODO Q: Should this check really be here? Feels like we should be sanitizing it regardless.
```

Comment:
@knsv Please do a <kbd>Ctrl + f</kbd> for `TODO Q:`. 
I've left multiple questions that we need to discuss.

Reply:
ok